install.js

```diff
@@ -0,0 +1,80 @@
+var os = require("os");
+var path = require("path");
+var ProgressBar = require("progress");
+var get = require("simple-get");
+var tar = require("tar");
+var ffmpegPath = require(".");
+var pkg = require("./package");
+
+function downloadExtractFile(url, destinationPath, progressCallback) {
+  let fulfill, reject;
+  let downloadedBytes = 0;
+  let totalBytes = 0;
+
+  const promise = new Promise((x, y) => {
+    fulfill = x;
+    reject = y;
+  });
+
+  get(url, function(err, response) {
+    if (err || response.statusCode !== 200) {
+      const error = new Error(`Download failed. URL: ${url}`);
+      reject(error);
+      return;
+    }
+
+    totalBytes = parseInt(response.headers["content-length"], 10);
+
+    response
+      .pipe(
+        tar.x({
+          C: path.dirname(destinationPath)
+        })
+      )
+      .on("finish", () => fulfill())
+      .on("error", error => reject(error));
+
+    if (progressCallback) {
+      response.on("data", function(chunk) {
+        downloadedBytes += chunk.length;
+        progressCallback(downloadedBytes, totalBytes);
+      });
+    }
+  });
+
+  return promise;
+}
+
+let progressBar = null;
+let lastDownloadedBytes = 0;
+function onProgress(downloadedBytes, totalBytes) {
```

Comment:
Just call `onProgress` with the delta (taken from `chunk.length`) instead of the total. This will remove unnecessary calculations forth and back.

Reply:
I need total bytes to instantiate the progress bar.

Reply:
You can change the signature to `onProgress(deltaBytes, totalBytes)`. Then you don't have to do calculations in two places.

Reply:
Got it 👍